tools/gen-stage-test-diff

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python3
+import argparse
+import contextlib
+import os
+import subprocess
+import tempfile
+
+
+def run_osbuild(output_directory, store, cache_max_size, libdir, manifest):
+    args = [
+        "osbuild",
+        "--export",
+        "tree",
+        "--output-directory",
+        output_directory,
+        "--store",
+        store,
+        "--cache-max-size",
+        str(cache_max_size),
+        "--checkpoint",
+        "tree",
+        "--checkpoint",
+        "build",
+    ]
+
+    if libdir:
+        args += ["--libdir", libdir]
+
+    args.append(manifest)
+    result = subprocess.run(
+        args,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        encoding="utf-8",
+        check=False,
+    )
+
+    if result.returncode != 0:
```

Comment:
If we're doing this might as well do `check=True` use the `CalledProcessError` (which also contains stdout/stderr.